airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/message/DestinationMessage.kt

```diff
@@ -318,6 +318,8 @@ data class GlobalCheckpoint(
     override val destinationStats: Stats? = null,
     val checkpoints: List<Checkpoint> = emptyList(),
     override val additionalProperties: Map<String, Any>,
+    val originalTypeField: AirbyteStateMessage.AirbyteStateType? =
+        AirbyteStateMessage.AirbyteStateType.GLOBAL,
```

Comment:
Is the goal here to capture whether type global was specified or it was left `null`? And is this field used anywhere?

Reply:
1 Yes. 2 It is used when the state message is pushed back to the platform.